### PR TITLE
Fixes #11906 - permissions

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

File: lib/foreman_deployments/engine.rb [hunk 1]
```diff
@@ -18,29 +18,39 @@ class Engine < ::Rails::Engine
...
+                         { :'foreman_deployments/api/v2/deployments' => [:index, :show] },
+                         :resource_type => ForemanDeployments::Deployment.name
           map.permission :create_deployments,
-                         :"foreman_deployments/api/v2/deployments" => [:create]
-          map.permission :edit_deployments,
-                         :"foreman_deployments/api/v2/deployments" => [:update]
+                         { :'foreman_deployments/api/v2/deployments' => [:create, :merge_configuration, :replace_configuration] },
+                         :resource_type => ForemanDeployments::Deployment.name
+          map.permission :run_deployments,
+                         { :'foreman_deployments/api/v2/deployments' => [:run] },
```
Comment (ares) [inline lib/foreman_deployments/engine.rb:30]: you'll need to define mapping for custom names in controllers, see https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/hostgroups_controller.rb#L84 for inspiration

**Comment by ares** [comment]
you have to add `.authrozied` scope in places where you need manually load data, e.g. https://github.com/theforeman/foreman_deployments/blob/master/app/controllers/foreman_deployments/api/v2/deployments_controller.rb#L22

**Comment by tstrachota** [comment]
@ares updated, may I ask you for one more brief check?

**Comment by ares** [comment]
LGTM

**Comment by tstrachota** [comment]
I'll rebase

**Comment by ShimShtein** [comment]
Merged as https://github.com/theforeman/foreman_deployments/commit/12fe180229f169e80dd7059340b43310b51fbfca
Thanks @tstrachota!

**Pull request closed without merging**